pandas/core/frame.py

```diff
@@ -374,8 +376,12 @@ def _constructor_expanddim(self):
     # ----------------------------------------------------------------------
     # Constructors
 
-    def __init__(self, data=None, index=None, columns=None, dtype=None,
-                 copy=False):
+    def __init__(self,
+                 data=None,
```

Comment:
Still need type for `data`, I think need a type for it in `_typing.py`. I will take a look at it later, meanwhile @jreback @WillAyd any suggestion will help.